spec/acceptance/qpid_spec.rb

```diff
@@ -1,6 +1,6 @@
 require 'spec_helper_acceptance'
 
-describe 'qpid' do
+describe 'qpid', :order => :defined do
```

Comment:
This is unfamiliar to me.

I would guess it ensures that the tests below run in the order they are defined, so that the `ensure => 'absent'` test is guaranteed to test removal of something that was already installed.

Is that correct and what would be the impact if this change were not included? Does this mean the other tests within that context (e.g. ensureing qpid-tools package is not installed) were not running in the correct order previously?

Are there other `ensure => 'absent'` like tests in our puppet module ecosystem, particularly related to your work about enabling and disabling katello-agent infrastructure, that should receive a similar change?

Thanks @ehelms ... any clarity you can provide around it is helpful.

Reply:
Exactly. The intent here was to try to enforce setting up qpid and then perform the `ensure => absent` with an installed qpid to ensure that everything worked as expected. The other way to do this would be to try and apply qpid within the absent test block independently. That just felt like overkill when I could use a defined order.